### PR TITLE
adding support for policy attach to bootstrap target role

### PR DESCRIPTION
**Pull request by dgraeber**
*Issue #, if available:*
#352 

*Description of changes:*
Adding ability to pass in an attach managed policies to deployment roles on bootstrap

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: docs/source/bootstrapping.md [hunk 1]
```diff
@@ -23,7 +24,12 @@ Options:
                                   deploy  [default: no-synth]
   --profile TEXT                  The AWS profile to initiate a session
   --region TEXT                   AWS region to use
-  --qualifier TEXT                A qualifier to append to toolchain role (alpha-numeric char max length of 6)
+  --qualifier TEXT                A qualifier to append to toolchain role
+                                  (alpha-numeric char max length of 6)
+  -pa, --policy-arn TEXT          ARN of existing Policy to attach to Target
```
Comment (srinivasreddych) [inline docs/source/bootstrapping.md:29]: can you make this more clear about it being valid in single account structure (`--as-target`)?

File: docs/source/bootstrapping.md [hunk 2]
```diff
@@ -54,7 +60,12 @@ Options:
                                   deploy  [default: no-synth]
   --profile TEXT                  The AWS profile to initiate a session
   --region TEXT                   AWS region to use
-  --qualifier TEXT                A qualifier to append to target role (alpha-numeric char max length of 6)
+  --qualifier TEXT                A qualifier to append to target role (alpha-
+                                  numeric char max length of 6)
+  -pa, --policy-arn TEXT          ARN of existing Policy to attach to Target
```
Comment (srinivasreddych) [inline docs/source/bootstrapping.md:65]: can you make this more clear about it being valid in single account structure (--as-target)?

File: seedfarmer/cli_groups/_bootstrap_group.py [hunk 3]
```diff
@@ -54,7 +54,8 @@ def bootstrap() -> None:
 @click.option(
     "--trusted-principal",
     "-t",
-    help="ARN of Principals trusted to assume the Toolchain Role",
+    help="""ARN of Principals trusted to assume the Toolchain Role.
```
Comment (srinivasreddych) [inline seedfarmer/cli_groups/_bootstrap_group.py:57]: can you make this more clear about it being valid in single account structure (--as-target)?

File: seedfarmer/cli_groups/_bootstrap_group.py [hunk 4]
```diff
@@ -100,11 +101,21 @@ def bootstrap() -> None:
     help="A qualifier to append to toolchain role (alpha-numeric char max length of 6)",
     required=False,
 )
+@click.option(
+    "--policy-arn",
+    "-pa",
+    help="""ARN of existing Policy to attach to Target Role (Deploymenmt Role)
```
Comment (srinivasreddych) [inline seedfarmer/cli_groups/_bootstrap_group.py:107]: can you make this more clear about it being valid in single account structure (--as-target)?

File: seedfarmer/cli_groups/_bootstrap_group.py [hunk 5]
```diff
@@ -117,10 +128,14 @@ def bootstrap_toolchain(
     if project is None:
         project = _load_project()
     _logger.debug("Bootstrapping a Toolchain account for Project %s", project)
+    if len(policy_arn) > 0 and not as_target:  # type: ignore
+        raise click.ClickException("Cannot set PolicyARNS and not set the -as-target flag")
```
Comment (srinivasreddych) [inline seedfarmer/cli_groups/_bootstrap_group.py:132]: rewrite: Cannot set PolicyARNS and when `-as-target` flag is not set?

File: docs/source/bootstrapping.md (hunk 1 above)
Comment (dgraeber) [reply docs/source/bootstrapping.md:29]: I can

File: seedfarmer/cli_groups/_bootstrap_group.py (hunk 5 above)
Comment (dgraeber) [reply seedfarmer/cli_groups/_bootstrap_group.py:132]: sure

File: seedfarmer/cli_groups/_bootstrap_group.py (hunk 3 above)
Comment (dgraeber) [reply seedfarmer/cli_groups/_bootstrap_group.py:57]: that doesn't apply to Principals on the Toolchain Role...disregarding this comment

File: seedfarmer/cli_groups/_bootstrap_group.py (hunk 4 above)
Comment (dgraeber) [reply seedfarmer/cli_groups/_bootstrap_group.py:107]: yes

File: docs/source/bootstrapping.md (hunk 2 above)
Comment (dgraeber) [reply docs/source/bootstrapping.md:65]: it doesn't apply here...disregarding this comment

**Review by srinivasreddych — APPROVED** [review]

**Pull request merged**